public/scripts/directives/ngFileSelect.js

```diff
@@ -3,7 +3,7 @@ app.directive("ngFileSelect",function(){
         link: function($scope,el){
             el.bind("change", function(e){
                 $scope.file = (e.srcElement || e.target).files[0];
-                $scope.getFile();
+                $scope.getFile($scope.file);
```

Comment:
You already have `file` in `$scope`, and you are also passing it as an argument to the function.

The function could use `$scope.file` or the argument passed, but having both is a little bit redundant. There should be only one responsible class of assigning $scope.file.